Sidebar compact issue1293

JInfantesC opened the pull request:
---
Issue #1293. Sidebar text labels and sizes
---
**Pull Request Type**
- [ ] Bugfix
- [X] Feature Implementation

**Related issue**
https://github.com/FreeTubeApp/FreeTube/issues/1293

**Description**
This commits revise the size of the icons in the sidebar and how the text labels displays using ellipsis on long texts, making use of CSS properties and the HTML title atributte.

**Screenshots (if appropriate)**
![Freetube_1293_ex1](https://user-images.githubusercontent.com/7262829/123556561-717e8b80-d77b-11eb-9790-a04fda329dc7.png)
![Freetube_1293_ex2](https://user-images.githubusercontent.com/7262829/123556563-72172200-d77b-11eb-8002-9b0cd5132d38.png)

**Testing (for code that is not small enough to be easily understandable)**
Has this pull request been tested?
As a CSS and HTML modification without touching anything more complex or critical, code changes are easily understandable. No test required

**Desktop (please complete the following information):**
 - OS: Ubuntu Linux
 - OS Version: 20.04
 - FreeTube version: 0.13.0

peepopoggers commented:
@JInfantesC link the pull request pls.

We were discussing this on the element matrix. I really think this is a good idea for compact mode.

JInfantesC commented:
Hi @peepopoggers, 

I'm afraid i don't understand what you mean by "link the pull request". Github does not allow me to use the "Linked issues" tool if that's what you mean.

Please, let me know what should i do. Thank you!

ChunkyProgrammer commented:
I think they want you to type "closes #1293" instead of "#1293" for the related issue of your description.

JInfantesC commented:
Thanks! Done!

ChunkyProgrammer commented on src/renderer/components/side-nav/side-nav.vue:
```diff
@@ -25,14 +30,19 @@
         class="navOption mobileHidden"
         role="button"
         tabindex="0"
+        :title="$t('Trending')"
```
This should be changed to `$t('Trending.Trending')` before merging (#1638 changed this)

ChunkyProgrammer commented:
Hi, I just tested your PR and it works fine for the most part.
It looks like this when using the mobile view though (I think "scr/renderer/components/side-nav-more-options" needs to be updated):
![image](https://user-images.githubusercontent.com/78101139/133308366-d01a1550-07dd-499d-872a-d1f9220e30a0.png)

This is what the mobile view looks like without using this PR:
![image](https://user-images.githubusercontent.com/78101139/133308665-ba37a002-a5ed-4271-8bdc-1f27c2400a06.png)

JInfantesC commented:
I've revised the more options component:
![Freetube more-options](https://user-images.githubusercontent.com/7262829/133835687-6ae69c2d-2d76-4cf7-b066-60a1e43cf4fe.png)

Thank you!

jasonhenriquez commented:
Getting rid of text labels & relying on the title attribute to provide the information is generally discouraged for accessibility reasons. I in fact believe that FreeTube should be moving toward the opposite direction. 
https://www.tpgi.com/html5-accessibility-chops-title-attribute-use-and-abuse/
https://www.mediacurrent.com/blog/dont-rely-title-attribute-accessibility-seo/
https://www.w3.org/WAI/tutorials/forms/labels/

PikachuEXE commented:
I agree with @jasonhenriquez.
I have seen UI with mostly icons only and it's really annoying and confusing on some of them.
I rather see text only buttons without the icons.
On PC I can still hover to see the title but annoying.
On Mobile there is no way to "hover".

efb4f5ff-1298-471a-8973-3d47447115dc commented:
Does this mean that we have to close the issue related to this?

ChunkyProgrammer commented:
https://github.com/FreeTubeApp/FreeTube/issues/1293#issuecomment-844996542
Should probably be an optional setting that is disabled by default?

PikachuEXE commented:
@ChunkyProgrammer 
It's that's how it's implemented I have no problem with that.
This PR is not implemented that way currently.

efb4f5ff-1298-471a-8973-3d47447115dc commented:
> https://github.com/FreeTubeApp/FreeTube/issues/1293#issuecomment-844996542
> Should probably be an optional setting that is disabled by default?

And maybe add something like a warning that if u enable it that it will not improve the UX for people with issues stated in comments above.

peepopoggers commented:
I agree that this should be an opt-in feature, but it is still cool for those who want a more minimal UI but have used FT enough to know their way round.

JInfantesC commented:
Thanks everyone for the comments!

I've encouraged myself to try this feature as an opt-in setting as discussed here.
The setting "Hide Compact Side Bar text" is available in settings/Distraction Free Settings. 

A few screenshots:
![sidenav_setting_2](https://user-images.githubusercontent.com/7262829/134573885-f7e36c2c-6f48-42f7-bf77-523b38cb9188.png)
![sidenav_setting_3](https://user-images.githubusercontent.com/7262829/134573898-5b394792-543f-4151-a357-8301b7abb526.png)

![sidenav_setting_4](https://user-images.githubusercontent.com/7262829/134573909-9c062fd1-a21f-4134-a89d-0c28c4f90418.png)
![sidenav_setting_5](https://user-images.githubusercontent.com/7262829/134573915-edb56af2-b419-4d31-ae29-fd7c091c2244.png)

Because this pull request is now more complex than initially defined, let me know if the changes made are going in the right direction.

PD. The `settings.js` instructions are wonderful and very helpful :)

Thank you!

jasonhenriquez commented:
@JInfantesC I appreciate your initiative in fixing this -- there are three things you will have to fix first.
1. Use a template string in en-us.yaml for the name of this control. This will allow it to be translatable into other languages.
2. Change the name to "Show Labels in Side Bar" throughout your code & as the value of the template string. As a result of this naming change, you will have to change the default value to "true" and reverse your logic for whether it should be displayed. The reason for this change is that the meaning of the original text is not very clear.
3. Move the toggle from Distraction Free Settings to Theme Settings. The reason for this change is that Distraction Free Settings are meant for hiding content that will distract someone. Labels are not distracting and more just explain the meaning of something.

After these changes, there should not be many additional changes left to make. Again, thank you so much for your good work.

ChunkyProgrammer commented:
> @JInfantesC I appreciate your initiative in fixing this -- there are three things you will have to fix first.
> 
> 1. Use a template string in en-us.yaml for the name of this control. This will allow it to be translatable into other languages.
> 2. Change the name to "Show Labels in Side Bar" throughout your code & as the value of the template string. As a result of this naming change, you will have to change the default value to "true" and reverse your logic for whether it should be displayed. The reason for this change is that the meaning of the original text is not very clear.
> 3. Move the toggle from Distraction Free Settings to Theme Settings. The reason for this change is that Distraction Free Settings are meant for hiding content that will distract someone. Labels are not distracting and more just explain the meaning of something. You will also have to put your template string in en-us.yaml under the Theme Settings group (seen on line 155 of en-us.yaml). Your label in the Vue code with look like this: `:label="$t('Settings.Theme Settings.Show Labels in Side Bar')"`
> 
> After these changes, there should not be many additional changes left to make. Again, thank you so much for your good work.

Should it not be "Hide Labels in Side Bar" iirc all the distraction free settings default to false so it'd be a bit odd having one default to true

peepopoggers commented:
I agree with Chunky here, it should be named "Hide labels in side bar" -> sounds better.

jasonhenriquez commented:
> Should it not be "Hide Labels in Side Bar" iirc all the distraction free settings default to false so it'd be a bit odd having one default to true

> > 3. Move the toggle from Distraction Free Settings to Theme Settings.

JInfantesC commented:
Thank you all for your comments. I've moved the switch to theme-settings and renamed it "Hide labels in Side Bar"

Please, take a look and do not hesitate to change anything if needed :)

peepopoggers commented on src/renderer/components/theme-settings/theme-settings.vue:
```diff
@@ -21,6 +21,11 @@
         @change="handleRestartPrompt"
       />
+      <ft-toggle-switch
+        :label="$t('Settings.Theme Settings.Hide labels in Side Bar')"
```
make it Hide Labels in Side Bar

peepopoggers commented:
Some feedback I want to give as I inspected your changes and tested the PR:

- In my opinion the new default (icon with text) icons are too big. I feel like they could be made a little bit smaller. (In no-label mode you can keep the icon sizes as you have changed them to be)
- You should also make the side bar a bit less wide when you hide the labels. That would make it more compact.

Other than that this is a great new addition

efb4f5ff-1298-471a-8973-3d47447115dc commented:
> In my opinion the new default (icon with text) icons are too big. I feel like they could be made a little bit smaller. (In no-label mode you can keep the icon sizes as you have changed them to be)

> You should also make the side bar a bit less wide when you hide the labels. That would make it more compact.

In my opinion both do not have to be addressed to me looks great the way it is.

efb4f5ff-1298-471a-8973-3d47447115dc reviewed (approved):
LGTM!

peepopoggers commented:
I would like some more feedback on my ideas before this gets merged

JInfantesC commented:
Thanks for the feedback @peepopoggers.

I've implemented those changes, the icons will keep the current size by default. When the hide label option is active those icons will get bigger.

And when the navbar is closed and the hide label option is active, the navbar width is set to 60px instead of 80px.

i've user computed properties and CSS selectors for these changes.

peepopoggers commented on src/renderer/components/theme-settings/theme-settings.vue:
```diff
@@ -21,6 +21,11 @@
         @change="handleRestartPrompt"
       />
+      <ft-toggle-switch
+        :label="$t('Settings.Theme Settings.Hide Labels in Side Bar')"
```
One final suggestion: Change to:
Hide Side Bar Labels

peepopoggers commented:
LGTM just 3 things

JInfantesC commented:
Done!
Also, i've fixed a getter on theme-settings.js.

peepopoggers reviewed (approved):
LGTM

ChunkyProgrammer reviewed (approved):
LGTM!

efb4f5ff-1298-471a-8973-3d47447115dc reviewed (approved):
LGTM!
